### PR TITLE
Fix clazy issues

### PR DESCRIPTION
**Pull request by sithlord48**
- Use Qt5 style connections 
    - Old SIGNAL / SLOT macros are slower and more error prone 
 - Use `const &` for current item and a `const list` for input to all ranged for loops to prevent detachment
 - Use `Q_EMIT` in place of `emit` emitting a signal (added a few missing ones also )

### CODE REVIEW

File: src/gui/src/ZeroconfService.cpp [hunk 1]
```diff
@@ -67,29 +67,24 @@ ZeroconfService::ZeroconfService(MainWindow* mainWindow) :
...
-        this, SLOT(errorHandle(DNSServiceErrorType)));
+    connect(zeroconf_browser_.get(), &ZeroconfBrowser::error, this, &ZeroconfService::errorHandle);
 }
 
 ZeroconfService::~ZeroconfService() = default;
 
 void ZeroconfService::serverDetected(const QList<ZeroconfRecord>& list)
 {
-    for (ZeroconfRecord record : list) {
+    for (const ZeroconfRecord &record : list) {
```
Comment (p12tic) [inline src/gui/src/ZeroconfService.cpp:87]: Prefer to place & with type.

File: src/gui/src/main.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ bool checkMacAssistiveDevices();
 
 void copy_qsettings(const QSettings &src, QSettings &dst)
 {
-    auto keys = src.allKeys();
+    const auto keys = src.allKeys();
```
Comment (p12tic) [inline src/gui/src/main.cpp:57]: There's no need for `const` for non-reference types. I've yet to hit a bug that `const` would have prevented, however it introduces a lot of visual noise. Little benefit with certain costs.

**Review by p12tic — COMMENTED** [review]
LGTM except two nitpicks. Thanks!

File: src/gui/src/main.cpp (hunk 2 above)
Comment (sithlord48) [reply src/gui/src/main.cpp:57]: This is due to a clazy check for detachment on the ranged for. We can't use `qAsConst` with the type so we make it `const` prevents the issue.
You can read it more about it [here](https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop-detach.md).

File: src/gui/src/ZeroconfService.cpp (hunk 1 above)
Comment (p12tic) [reply src/gui/src/ZeroconfService.cpp]: Also in other places :-)

Comment (p12tic) [reply src/gui/src/ZeroconfService.cpp]: Fixed.

**Review by p12tic — APPROVED** [review]

**Pull request merged**